### ISSUE TITLE
Existing metadata key values are not populated in entry form

### ISSUE THREAD

**Issue by mgardnerpsu** [Duke-Translational-Bioinformatics/duke-data-service-portal#556]
When entering template/key values for a file - existing key values are not populated - they render an empty form field.  In addition, if a new value is provided the and saved - when returned to the file detail page - the old value is still displayed.

**Comment by caseychoiniere** [comment]
@mgardnerpsu I'm going to split this into 2 separate issues. 

I've fixed the issue with the page not updating with the new metadata values.

The problem with template key values not populating is a lot more complex. Although this would normally be an easy fix, due to the complexity of the metadata service it isn't so easy in this case. Basically when you are viewing a template and it's keys you are not viewing in the context of a Metadata object but in the context of a template property instance. 

Kind of confusing but the flow goes like this:

1.) User selects the "Add advanced metadata" option

2.) API call is made to get all templates.

3.) User selects a template to use

4.) API call made to get the template properties (no values here because this is in the context of a template property, not a metadata object)

5.) List of template properties is rendered in the UI, each having the appropriate form control for their datatype.

6.) The values for each template property exist in the context of the metadata object, not the template property. 

I tried getting this to work when I was building the feature a few months ago but could never get it to work.

So the crux lies in the fact that in order to populate those fields, the UI needs to separately parse all of the current Metadata objects for the file and then determine whether a template property in the list has a template_property ID that matches any of the template_property ID's included in any available metadata objects. And then check that template_property to see if it has a value. And all of this has to happen before the UI renders (preferably).  Which is tough when working with two asynchronous actions like this. I also remember there was an additional challenge due to the way the Object Metadata is returned. It contains an array of property objects that all have a `value` property and a `template property` property (object).   Anyway, thinking out loud here, this is probably one of those things that when I come back to it the answer will be obvious. 
```javascript
{
    "object": {
      "kind": "dds-file",
      "id": "5ce6ec9a-011f-40ea-bc8a-cf28f83ef231"
    },
    "template": {
      "id": "43a4365d-b9ae-48df-9211-0d237c25fb8f",
      "name": "test_ohsdfhiooihwdfshiohiou"
    },
    "properties": [
      {
        "value": "12",
        "template_property": {
          "id": "f56c3674-af59-4370-ad48-eb12ba5e36cf",
          "key": "test_really_long_keytest_really_long_keytest_really_long_key",
          "label": "test_really_long_key"
        }
      },
      {
        "value": "2017-01-27",
        "template_property": {
          "id": "6c3846de-caf9-4fd9-b491-3ce69b9bb4ba",
          "key": "test_prop_1273198237897891278",
          "label": "test_prop_1273198237897891278"
        }
      },
```

**Comment by mgardnerpsu** [comment]
So if we added `properties[].template_property.type` to this payload it seems life would be easier...

3.) User selects a template to use. 

3.1) API call gets made to see if template has been associated to this object already i.e. `GET /meta/{object_kind}/{object_id}/{template_id}` - if so, properties (and their type and optional value) returned.

[only if 3.1 returns 404 not found] 4.) API call made to get the template properties (no values here because this is in the context of a template property, not a metadata object)

5.) List of properties is rendered in the UI, each having the appropriate form control for their `type` - if `value` exists it is populated.

**Comment by caseychoiniere** [comment]
I have it figured out using the existing services. A bit of data manipulation in the client store, some filtering and I can make it work fine. 

What you propose would certainly make it easier but I'm not sure that it's worth the resources.

I'm glad you logged this though. It's one of those problems I just needed to come back to later and then the solution was fairly obvious.

**Comment by caseychoiniere** [comment]
fixed https://github.com/Duke-Translational-Bioinformatics/duke-data-service-portal/pull/559
